components/bitcoin-topics/markdown-ui/script-visualizer/scriptVisualizer.tsx

```diff
@@ -1,272 +1,336 @@
+                })
             }
-        })
-    }
+        }
+    }, [config])
 
     useEffect(() => {
-        const timer = setTimeout(() => handlePlayerAction("play"), 100)
+        if (!config) return
+
+        const timer = setTimeout(() => {
+            initializeSvgPlayer()
+        }, 100)
         return () => clearTimeout(timer)
-    }, [svgIndex])
+    }, [initializeSvgPlayer, config])
 
-    const handleSvgSelect = (index: number) => {
-        setSvgIndex(index)
-    }
+    const playStepAnimation = useCallback(
+        (step: number) => {
+            if (!config || !playerRef.current) return
 
-    const handleSvgSwitch = (direction: "next" | "prev") => {
-        setSvgIndex((prev) => {
-            const newIndex =
-                direction === "next"
-                    ? (prev + 1) % svgPaths.length
-                    : (prev - 1 + svgPaths.length) % svgPaths.length
-            return newIndex
-        })
-    }
+            if (animationTimeoutRef.current) {
+                clearTimeout(animationTimeoutRef.current)
+            }
+
+            const startTime = step > 0 ? cumulativeDurations[step - 1] : 0
+            const endTime = cumulativeDurations[step]
+            playerRef.current.seekTo(startTime)
+            playerRef.current.play()
+            setIsPlaying(true)
+
+            const animationDuration = endTime - startTime
+            animationTimeoutRef.current = setTimeout(() => {
+                playerRef.current?.pause()
+                setIsPlaying(false)
+            }, animationDuration)
+        },
+        [config, cumulativeDurations]
+    )
+
+    const handlePlay = useCallback(() => {
+        if (!isPlaying && currentStep >= 0) {
+            playStepAnimation(currentStep)
+        }
+    }, [isPlaying, currentStep, playStepAnimation])
+
+    const handlePause = useCallback(() => {
+        if (playerRef.current) {
+            playerRef.current.pause()
+            setIsPlaying(false)
+            if (animationTimeoutRef.current) {
+                clearTimeout(animationTimeoutRef.current)
+            }
+        }
+    }, [])
+
+    const handleStepChange = useCallback(
+        (direction: "next" | "prev") => {
+            if (!config) return
+
+            setCurrentStep((prev) => {
+                const newStep =
+                    direction === "next"
+                        ? Math.min(prev + 1, config.steps.length - 1)
+                        : Math.max(prev - 1, 0)
+                if (newStep >= 0) {
+                    playStepAnimation(newStep)
+                }
+                return newStep
+            })
+        },
+        [config, playStepAnimation]
+    )
+
+    const handleStepClick = useCallback(
+        (index: number) => {
+            setCurrentStep(index)
+            playStepAnimation(index)
+        },
+        [playStepAnimation]
+    )
 
-    function classNames(...classes: string[]) {
+    function classNames(...classes: string[]): string {
```

Comment:
I've pointed this out in your previous PRs but possibly you may have skipped as there are alot of reviews but I'll appreciate it if you make this change.
Can you add this `classNames` function to the utils files and use it from there as it is been used in different sections across the codebase.

Reply:
done